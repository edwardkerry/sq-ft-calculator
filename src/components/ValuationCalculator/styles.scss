.valuation {
  height: 20%;
  margin-top: 20vh;
  padding: 10px;
  border: 2px solid rgb(53, 58, 57);
  background-color: rgb(169, 170, 177);
  justify-content: center;
}

.price {
  font-size: 20px;
  color: rgb(113, 228, 113);
}
