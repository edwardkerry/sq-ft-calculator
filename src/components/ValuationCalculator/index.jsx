import React, { Component } from 'react';
import { arrayOf, shape, number } from 'prop-types';

import './styles.scss';

class ValuationCalculator extends Component {

  constructor(props) {
    super(props)

    this.state = {
      value: 0
    }

    this.getAllFootageCosts = this.getAllFootageCosts.bind(this)
    this.getAverageFootageCost = this.getAverageFootageCost.bind(this)
    this.calculateValue = this.calculateValue.bind(this)
  }

  componentDidMount() {
    this.setState({
      value: this.calculateValue()
    })
  }

  getAllFootageCosts() {
    return this.props.properties.map((property) => {
      return Math.round(property.price / property.floor_area)
    })
  }

  getAverageFootageCost() {
    const allFootageCosts = this.getAllFootageCosts()
    const reducer = (accumulator, currentValue) => accumulator + currentValue;
    return Math.round(allFootageCosts.reduce(reducer) / allFootageCosts.length)
  }

  calculateValue() {
    const averageCostPerFoot = this.getAverageFootageCost()
    return averageCostPerFoot * 1100
  }

  render() {
    return (
      <div className='valuation'>
        A 1,100 sq ft property will cost
        <p className='price'>Â£{this.state.value} </p>
      </div>
    )
  };
}

ValuationCalculator.propTypes = {
  properties: arrayOf(
    shape({
      price: number.isRequired,
      floor_area: number.isRequired
    }).isRequired
  ).isRequired
};

export default ValuationCalculator;
