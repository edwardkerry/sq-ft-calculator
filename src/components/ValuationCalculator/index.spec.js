import React from 'react';
import renderer from 'react-test-renderer';
import ValuationCalculator from './';

describe('Valuation Calculator', () => {

  const properties = [
    {
      price: 300000,
      floorArea: 300
    },{
      price: 500000,
      floorArea: 500
    },{
      price: 1000000,
      floorArea: 500
    }
  ]

  it('renders', () => {
    const component = renderer.create(< ValuationCalculator properties={properties} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  });

  const wrapper = shallow(<ValuationCalculator properties={properties} />).instance()

  describe('#getAllFootageCosts', () => {
    it('returns an array of price per square foot per property', () => {
      const expectedResult = [1000, 1000, 2000]
      expect(wrapper.getAllFootageCosts()).toEqual(expectedResult)
    })
  });

  describe('#getAverageFootageCost', () => {
    it('returns a rounded average of all numbers', () => {
      const expectedResult = 1333
      expect(wrapper.getAverageFootageCost()).toEqual(expectedResult)
    })
  });

  describe('#caculateCalue', () => {
    it('returns the price of a 1,100 sq ft property based on average', () => {
      const expectedResult = 1466300
      expect(wrapper.calculateValue()).toEqual(expectedResult)
    })
  });

});
