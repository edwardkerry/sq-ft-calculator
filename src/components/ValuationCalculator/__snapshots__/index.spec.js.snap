// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Valuation Calculator renders 1`] = `
<div
  className="valuation"
>
  A 1,100 sq ft property will cost
  <p
    className="price"
  >
    Â£
    1466300
     
  </p>
</div>
`;
